tests/test_nodes.py

```diff
@@ -2,19 +2,54 @@
 from decimal import Decimal, getcontext
 from http import HTTPStatus
 import json
+from typing import Any, Callable
 import urllib3
 
 import warnings
 
 
-def test_nodes():
+def validate_rpc(rpc: dict[str, Any], method: str, validator: Callable[[dict[str, Any]], None]) -> None:
+    """Validate RPC endpoint with specified method and validation function"""
+    try:
+        resp = urllib3.request(
+            method='POST',
+            url=(url := rpc['endpoint']),
+            json={'method': method, 'params': [], 'id': 1, 'jsonrpc': '2.0'},
+            retries=1,
+            timeout=5,
+        )
+    except urllib3.exceptions.MaxRetryError as e:
+        warnings.warn(f'Failed to connect to {url} due to {e}')
+        return
+
+    if resp.status in (HTTPStatus.TOO_MANY_REQUESTS, HTTPStatus.GATEWAY_TIMEOUT):
+        warnings.warn(f'Failed to connect to {url} due to {resp.status}')
+        return
+    
+    try:
+        validator(resp.json()) is True
```

Comment:
```suggestion
        assert validator(resp.json()) is True
```
needs to assert this